specification/help/Help.Management/DiagnosticResource.tsp

```diff
@@ -0,0 +1,80 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+namespace Microsoft.Help;
+/**
+ * Diagnostic resource
+ */
+model DiagnosticResource
+  is Azure.ResourceManager.ExtensionResource<DiagnosticResourceProperties> {
+  ...ResourceNameParameter<
+    Resource = DiagnosticResource,
+    KeyName = "diagnosticsResourceName",
+    SegmentName = "diagnostics",
+    NamePattern = "^[A-Za-z0-9-+@()_]+$"
+  >;
+}
+
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-interface-requires-decorator" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
```

Comment:
Please fix all 'FIXME' comments in the PR by replacing them with real suppression reasons, or eliminating the suppression.